src/aics-image-viewer/components/App/index.tsx

```diff
@@ -601,343 +384,148 @@ export default class App extends React.Component<AppProps, AppState> {
-          newSelectionState.renderMode = RenderMode.volumetric;
-          this.onChangeRenderingAlgorithm(RenderMode.volumetric);
-        }
-      }
-    } else if (
-      viewerSettings.viewMode !== ViewMode.threeD &&
-      this.state.viewerSettings.imageType === ImageType.segmentedCell
-    ) {
-      // switching from 2D to 3D
-      newSelectionState.maskAlpha = ALPHA_MASK_SLIDER_3D_DEFAULT;
-    }
 
-    this.handleChangeViewerSetting("viewMode", newMode);
-    if (newSelectionState.maskAlpha !== undefined) {
-      this.handleChangeViewerSetting("maskAlpha", newSelectionState.maskAlpha);
+    const aimg = new Volume(rawDims);
+    const volsize = rawData.shape[1] * rawData.shape[2] * rawData.shape[3];
+    for (var i = 0; i < rawDims.channels; ++i) {
+      aimg.setChannelDataFromVolume(i, new Uint8Array(rawData.buffer.buffer, i * volsize, volsize));
     }
-    Object.keys(newSelectionState.region!).forEach((axis) => {
-      const [minval, maxval] = newSelectionState.region![axis as AxisName];
-      this.handleImageAxisClipUpdate(axis as AxisName, minval, maxval, axis === activeAxis);
-    });
-    this.setViewerSettingsInState(newSelectionState);
-  }
-
-  onUpdateImageMaskAlpha(sliderValue: number): void {
-    this.setViewerSettingsInState({ maskAlpha: sliderValue });
-  }
 
-  onAutorotateChange(): void {
-    this.setViewerSettingsInState({
-      autorotate: !this.state.viewerSettings.autorotate,
+    let channelSetting = rawDims.channel_names.map((channel, index) => {
+      let color = (INIT_COLORS[index] ? INIT_COLORS[index].slice() : [226, 205, 179]) as ColorArray; // guard for unexpectedly longer channel list
+      return initializeOneChannelSetting(aimg, channel, index, color);
     });
-  }
-
-  private handleImageAxisClipUpdate(axis: AxisName, minval: number, maxval: number, isOrthoAxis: boolean): void {
-    const { view3d, image } = this.state;
-    if (view3d && image) {
-      view3d.setAxisClip(image, axis, minval - 0.5, maxval - 0.5, isOrthoAxis);
-    }
-  }
 
-  setImageAxisClip(axis: AxisName, minval: number, maxval: number, isOrthoAxis: boolean): void {
-    const { region } = this.state.viewerSettings;
-    this.setViewerSettingsInState({ region: { ...region, [axis]: [minval, maxval] } });
-    this.handleImageAxisClipUpdate(axis, minval, maxval, isOrthoAxis);
-  }
+    // Here is where we officially hand the image to the volume-viewer
+    placeImageInViewer(aimg, channelSetting);
 
-  makeUpdatePixelSizeFn(i: number): (value: number) => void {
-    const { pixelSize } = this.props;
-    const imagePixelSize = pixelSize ? pixelSize.slice() : [1, 1, 1];
-    return (value: number) => {
-      const pixelSize = imagePixelSize.slice();
-      pixelSize[i] = value;
-      this.state.image?.setVoxelSize(pixelSize);
-    };
-  }
+    setImage(aimg);
+    setChannelGroupedByType(makeChannelIndexGrouping(rawDims.channel_names, props.viewerChannelSettings));
+    setChannelSettings(channelSetting);
```

Comment:
is this sequence of {placeImageInViewer, setImage, ..., setChannelSettings etc} repeated a couple times?

Reply:
for example, is setImage ever called without a call to placeImageInViewer?

Reply:
It doesn't look to be, though this pattern only appears in two places for both ways of getting an image into the viewer (url and raw data), and those call sites don't have any state setters besides `setImage` in common. Nonetheless, I could see moving `setImage` into `placeImageInViewer` because you'll never not want to do both.